aws/resource_aws_ec2_transit_gateway_vpc_attachment.go

```diff
@@ -22,6 +22,15 @@ func resourceAwsEc2TransitGatewayVpcAttachment() *schema.Resource {
 		},
 
 		Schema: map[string]*schema.Schema{
+			"appliance_mode_support": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  ec2.ApplianceModeSupportValueDisable,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.ApplianceModeSupportValueDisable,
+					ec2.ApplianceModeSupportValueEnable,
+				}, false),
```

Comment:
Nit: In the future, this will be reported by a linter. Reference: https://github.com/hashicorp/terraform-provider-aws/issues/14601

```suggestion
				Type:         schema.TypeString,
				Optional:     true,
				Default:      ec2.ApplianceModeSupportValueDisable,
				ValidateFunc: validation.StringInSlice(ec2.ApplianceModeSupportValue_Values(), false),
```